### ISSUE TITLE
Layout/SpaceBeforeBrackets generates invalid code

### ISSUE THREAD

**Issue by adfoster-r7** [rubocop-hq/rubocop#9499]
### Input

```ruby
foo = "hello world!"
foo[  6, 5]  = "rubocop"
foo
```

### Command
```
$ rubocop -a --only Layout/SpaceBeforeBrackets foo.rb
Inspecting 1 file
C

Offenses:

foo.rb:2:4: C: [Corrected] Layout/SpaceBeforeBrackets: Remove the space before the opening brackets.
foo[  6, 5]  = "rubocop"
   ^^

```

###Output 
```ruby
foo = "hello world!"
foo 6, 5]  = "rubocop"
foo
```

The code is now invalid:
```ruby
ruby foo.rb 
foo.rb:2: syntax error, unexpected ']', expecting end-of-input
foo 6, 5]  = "rubocop"
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
1.9.1 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 2.7.2 x86_64-darwin19)
```
